packages/react/src/NavList/NavList.tsx

```diff
@@ -288,6 +290,54 @@ const Group: React.FC<NavListGroupProps> = ({title, children, sx: sxProp = defau
   )
 }
 
+export type NavListShowMoreItemProps = {
+  children: React.ReactNode
+  label: string
+} & SxProp
+
+const ItemWithinGroup = React.createContext<{groupId: string} | null>(null)
+
+const ShowMoreItem = React.forwardRef<HTMLButtonElement, NavListShowMoreItemProps>(
+  ({label, children, ...props}, forwardedRef) => {
+    const [expanded, setExpanded] = React.useState(false)
+    const targetFocused = React.useRef(false)
+
+    const groupId = useId()
+
+    React.useEffect(() => {
+      if (expanded && !targetFocused.current) {
+        const focusTarget: HTMLAnchorElement | null = document.querySelector(`[data-show-more-group-id="${groupId}"]`)
```

Comment:
Will this always be an `HTMLAnchorElement` or could it be another type? Couldn't quite tell from `NavList` if it forces `<a>` or not so wanted to double-check 👀

Reply:
I believe so. We don't seem to force it as `NavList.Item as="button"` works, but it's not officially supported (e.g. styles will most likely be incorrect if it's anything but a link `<a>`). I _think_ it's a safe bet to rely on the `HTMLAnchorElement`, but we could make it more generic!